applications/rag/frontend/container/rai/nlp_filter.py

```diff
@@ -0,0 +1,72 @@
+# Copyright 2024 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import os
+from google.cloud import language_v1
+from . import retry
+
+# Convert the project id into a full resource id.
+parent = os.environ.get('PROJECT_ID', 'NULL')
+
+# Instantiate a nlp client.
+nature_language_client = language_v1.LanguageServiceClient()
+
+def is_nlp_api_enabled():
+  if parent == 'NULL':
+    return False
+  # Check if the DLP API is enabled
+  try:
+    sum_confidences("test")
+    return True
+  except Exception as e:
+    print(f"Error: {e}")
+    return False
+
+def sum_confidences(text):
```

Comment:
Comments for these functions - especially sum_confidences - which is not clear what it does from its name.